packages/commonwealth/server/migrations/20230217164706-root-id-refactor.js

```diff
@@ -0,0 +1,134 @@
+'use strict';
+
+// The purpose of this script is to turn root_id to thread_id. To achieve this without losing any data, we must set the
+// comments that pointed to a proposal, to instead point to a thread that is created to backlink to that proposal
+module.exports = {
+  up: async (queryInterface, Sequelize) => {
+    await queryInterface.sequelize.transaction(async (t) => {
+      const updateQueries = [];
+
+      const threadComments = await queryInterface.sequelize.query(
+        `SELECT * FROM "Comments" WHERE "root_id" ILIKE '%discussion%'`
+      );
+
+      // for each comment that is part of a discussion, replace its root_id with the thread_id
+      threadComments[0].forEach((c) => {
+        updateQueries.push(
+          queryInterface.sequelize.query(
+            `UPDATE "Comments" SET "root_id" = ${
+              c.root_id.split('_')[1]
+            } WHERE "id"=${c.id}`,
+            { transaction: t }
+          )
+        );
+      });
+
+      // we will now perform the same operation with proposal comments, but in order to do so we must create threads
+      // to backlink to proposals
+      const proposalComments = await queryInterface.sequelize.query(
+        `SELECT * FROM "Comments" WHERE "root_id" NOT ILIKE '%discussion%'`
+      );
+
+      // maps root_id -> newThread it creates
+      const newThreads = new Map();
+      proposalComments[0].forEach((c) => {
+        if (newThreads.get(c.root_id)) {
+          return; //if we already had created thread to backlink proposal comments, don't create another one
+        }
+
+        let rootParts = c.root_id.split('_');
+        // handle cosmosproposals differently
+        if (rootParts[0] === 'cosmosproposal') {
```

Comment:
Can you check on aave (dydx) if we need to use this case there as well?

Reply:
So I am not sure what to look for since the aave networks don't have any special proposal type. Where would I look to find this proposal type?

Reply:
See comment id 13333 root_id "onchainproposal_52" chain aave

Reply:
So firstly thank you for catching this I will add it to the conditional.

But just for clarification, I am a little confused how the comments for this root_type works because:
/aave/proposal/52-renew-aave-grants-dao (which I think should correspond to this comment?) doesn't seem to have this comment in it on master or prod. This change will link this archive thread to this link, if this is not the correct place let me know.

Reply:
Perhaps the comment got deleted -- I just looked in my database for comments on proposals but didn't check on the site. Linking to the archive thread is perfect.